framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -101,9 +103,23 @@
                 infer(typeFactory, argTypes, assignedTo, methodElem, methodType, targets);
 
         handleUninferredTypeVariables(typeFactory, methodType, targets, inferredArgs);
+        // Type arguments cannot be primitives, so box them if they are.
+        boxPrimitives(typeFactory, inferredArgs);
```

Comment:
Do you agree that this is a fix-up to a problem, not the right solution?
To me, this is an ugly hack. The right thing is to correctly box arguments and assignment context before inference even begins. Doing this hack afterwards might solve the immediate problem, but we should at least clearly document that this is not the right solution.